src/Mado/QueryBundle/Queries/QueryBuilderFactory.php

```diff
@@ -39,60 +40,9 @@ class QueryBuilderFactory extends AbstractQuery
 
     protected $select;
 
-    /** @todo move this file in configuration */
-    /** @todo type number|text */
-    private static $operatorMap = [
-        //'eq' => [
-            //'filtro' => '=',
-            //'tipo' => 'data|numero|stringa',
-            //'meta' => '%{foo}%'
-        //],
-        'eq' => [
-            'meta' => '=',
-        ],
-        'neq' => [
-            'meta' => '!=',
-        ],
-        'gt' => [
-            'meta' => '>',
-        ],
-        'gte' => [
-            'meta' => '>=',
-        ],
-        'lt' => [
-            'meta' => '<',
-        ],
-        'lte' => [
-            'meta' => '<=',
-        ],
-        'startswith' => [
-            'meta' => 'LIKE',
-            'substitution_pattern' => '{string}%'
-        ],
-        'contains' => [
-            'meta' => 'LIKE',
-            'substitution_pattern' => '%{string}%'
-        ],
-        'notcontains' => [
-            'meta' => 'NOT LIKE',
-            'substitution_pattern' => '%{string}%'
-        ],
-        'endswith' => [
-            'meta' => 'LIKE',
-            'substitution_pattern' => '%{string}'
-        ],
-        'list' => [
-            'meta' => 'IN',
-            'substitution_pattern' => '({string})',
-        ],
-        'field_eq' => [
-            'meta' => '=',
-        ],
-    ];
-
     public function getAvailableFilters()
     {
-        return array_keys(static::$operatorMap);
+        return Operators::getOperators();
```

Comment:
Operators::getOperators() returns the operators's map and array_keys(static::$operatorMap) returns array_keys of operators's map. Maybe a test here can clarify this behavior.

Reply:
Yep I'm not sure about this change because I don't understand why before this pr there is array_keys but we need to check this together, I add a new simple test but the behavior needs to clarify well I think

Reply:
Operators::getOperators(); returns "self::$operatorMap".

QueryBuilderFactory::getAvailableFilters() must return `array_keys(static::$operatorMap)`. This means the only legal change here could be `return array_keys(Operators::getOperators())`.

A good improvemente here could be Operators::getKeys().

This means that getAvailableFilters returs alle the keys of operators. For some "legacy" reasons filters and operators are the same thing. In version 3, we can remove one or the other. While we are in 2.0 development we can decide to deprecate one of them to remove

Reply:
Ok I have understand the mistake, before this PR the function ```getAvailableFilters``` had never been called from the code.
Inside the code there was this directly to get operator array not the function:
```
self::$operatorMap[self::DEFAULT_OPERATOR]
```

So It's not necessary to return ```array_keys``` inside  ``getAvailableFilters```.

What do you think?

Reply:
We must maintain backward compatibility https://en.wikipedia.org/wiki/Backward_compatibility. But we can also improve api. If you think a method should be removed, just mark it as deprecated. It will be removed in netx major release.

Reply:
Maybe is not necessary but if you change api or values returned, ... you introduce a backward compatibility.